source/elements/oneMKL/source/domains/dft/enums_and_config_params.rst

```diff
@@ -152,29 +153,31 @@ config_param
+      * -   :ref:`BWD_STRIDES<onemkl_dft_config_data_layouts>`
+        -   Defines the data layout in the backward domain. The value for a :math:`d`-dimensional transform is a :math:`(d+1)`-long vector of type ``std::vector<std::int64_t>`` containing the offset and stride(s) for the backward-domain data. See :ref:`onemkl_dft_config_data_layouts` for more details and default values.
+      * -   :ref:`INPUT_STRIDES<onemkl_dft_config_data_layouts>` (deprecated)
+        -   Defines the input data layout. The value for a :math:`d`-dimensional transform is a :math:`(d+1)`-long vector of type ``std::vector<std::int64_t>`` containing the offset and stride(s) for the input data. See :ref:`onemkl_dft_config_data_layouts` for more details and default values
+      * -   :ref:`OUTPUT_STRIDES<onemkl_dft_config_data_layouts>` (deprecated)
+        -   Defines the output data layout. The value for a :math:`d`-dimensional transform is a :math:`(d+1)`-long vector of type ``std::vector<std::int64_t>`` containing the offset and stride(s) for the output data. See :ref:`onemkl_dft_config_data_layouts` for more details and default values
+      * -   :ref:`FWD_DISTANCE<onemkl_dft_config_data_layouts>`
+        -   If computing multiple (batched) transforms, this parameter specifies the distance (in elements) between the first data elements of consecutive data sets in the forward domain. Provided in type ``std::int64_t``, the default value is 0.
+      * -   :ref:`BWD_DISTANCE<onemkl_dft_config_data_layouts>`
+        -   If computing multiple (batched) transforms, this parameter specifies the distance (in elements) between the first data elements of consecutive data sets in the backward domain. Provided in type ``std::int64_t``, the default value is 0.
       * -   WORKSPACE
         -   Some FFT algorithm computation steps require a scratch space for permutations or other purposes.  To manage the use of auxiliary storage, set to ``config_value::ALLOW`` to permit the use of auxiliary storage and ``config_value::AVOID`` to avoid using auxiliary storage if possible.
       * -   ORDERING
         -   Some FFT algorithms apply an explicit permutation stage that can be time consuming.  The value of ``config_value::ORDERED`` (default) applies the data ordering for all transformations.  The value of ``config_value::BACKWARD_SCRAMBLED`` applies ordering for forward transform, but allows backward transform to have scrambled data if it gives a performance advantage. 
       * -   TRANSPOSE
         -   A boolean value to indicate providing the transposition of output results (for multi-dimensional transforms). Default value is ``false``.
-      * -   :ref:`PACKED_FORMAT<onemkl_dft_conjugate_even_storage>`
```

Comment:
Please note: got rid of more in the end: ``ORDERING`` is gone (even the corresponding original C developer references are extremely vague about that one), ``TRANSPOSE`` is gone (that would be specific to distributed DFTs actually, in my understanding). I also considered removing ``WORKSPACE`` as I find that one rather misnamed and possibly misleading for people who might be looking for "external workspace" feature as was enabled in closed-source oneMKL a year ago. @hparks-intel & @dbollapr, what would your opinion be about that ``config_param::WORKSPACE``? Keep it or leave it?

Reply:
I'd vote to leave it. Some backends already implement something similar. As discussed offline, this parameter is about whether to use a workspace at all, regardless of internal or external, so the un-decorated name seems appropriate enough to me.

Reply:
Actually, upon reviewing the current implementation and [documentation of the external workspace feature in closed-source oneMKL](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2023-2/user-allocated-workspaces.html#USER-S-INTENTION-REGARDING-WORKSPACE), it looks like that very same value of ``config_param::WORKSPACE`` is used therein (with possible corresponding configuration values ``config_value::WORKSPACE_{IN,EX}TERNAL``) to notify if users want to pass an external workspace or not. I don't see a way to cover the two (_different_) features with the same unique configuration parameter. Therefore, I don't see a way to make the closed-source oneMKL lib spec-compliant in the future unless we either 1) remove this from the specs while we still can  or 2) face the ugly realization that we must change the current implementation of the external workspace feature in closed-source oneMKL to be spec-compliant for enabling a(nother) niche feature that's implemented only for a few CPU code paths (in my understanding).

Reply:
I think it might be best to leave the `WORKSPACE` as is for now. @hjabird has just started looking into another specification modification to address that.
Are you aiming to implement this specification with the closed source oneMKL?

Reply:
@FMarno, my understanding is that we will be forced to, "some time soon", and from @hjabird's email from this morning, it seems like a different feature is indeed targeted by that on-going work. Personally, I don't see how to target two separate features (1. allow (resp. avoid) using a workspace at all (resp. if possible), 2. use an internally-/externally-allocated workspace, if using one) with one single configuration parameter unless we expand the possible configuration values to a (new) set of 4 possible values like `(IN|EXT)ERNAL_(ALLOW|AVOID)` to make. I find that rather messy and my personal preference would be to have two separate, adequately-named configuration parameters targeting either feature. So my personal preference would be to remove this one for now and have a separate (targeted) discussion when @hjabird has a proposal.

Reply:
We are indeed aiming to implement this specification with the closed source oneMKL, and we're aware that how we currently implement some non-spec external workspace functionality conflicts with our ability to implement this cleanly. I might argue that Hugh's current work on bringing that external workspace functionality to the spec is actually another reason to remove this for now. I don't think it's used in it's current form, and we can bring it back in a more robust form that accommodates both allow/avoid and internal/external workspace distinctions. But really either way is fine with me. Do you think we need to keep it because it's currently being used somewhere?

Reply:
I have no evidence that the `WORKSPACE` parameter is being used anywhere. We should potentially add a check to some of the backends in oneMKL to ensure it is set to allow at all times. I guess that means it should probably be removed for now.

WRT representing "targetting two separate features", one option could be to just have 3 options: `AVOID`, `INTERNAL`, and `EXTERNAL`. Setting `WORKSPACE` to `INTERNAL` or `EXTERNAL` would imply that workspaces are allowed, since any other configuration would be invalid. We can leave that to @hjabird's proposal though.

Reply:
One aspect of the ``AVOID`` value is the "if possible" in the description of the corresponding configuration parameter as it stands in the current (and this revised) form of the specs: an implementation may still have an (unavoidable) need for a workspace even if that was set to ``AVOID`` and, conceptually, one may want that unavoidable workspace allocated internally or externally...

I've removed it (as well as the corresponding `config_value::(ALLOW|AVOID)`) in [9e336a2f8707b4a85cdc11e6b404a49b54960b34](https://github.com/oneapi-src/oneAPI-spec/pull/506/commits/9e336a2f8707b4a85cdc11e6b404a49b54960b34), this needs better design and a dedicated discussion, in my opinion.

Reply:
In https://github.com/oneapi-src/oneAPI-spec/pull/509, I've suggested
* `WORKSPACE_PLACEMENT` with values `WORKSPACE_AUTOMATIC` and `WORKSPACE_EXTERNAL`
* `WORKSPACE` can still be set to `ALLOW` and `AVOID`.

Reply:
cuFFT has two different APIs, `cufftResult cufftSetAutoAllocation(cufftHandle plan, int autoAllocate);`,  they basically get away with internal or external workspace usage with just an `int`. And they have `cufftResult cufftXtSetWorkAreaPolicy(cufftHandle plan, cufftXtWorkAreaPolicy policy, size_t *workSize);`. that sets the few workspace policies among which, `CUFFT_WORKAREA_MINIMAL` -> this I believe translates to our `AVOID` option. So we might need two different enums for this purpose as @hjabird  mentioned(as a side note I prefer `WORKSPACE_PLACEMENT` -> `WORKSPACE_ALLOCATION` ).

Reply:
Does anyone object to removing `config_param::WORKSPACE` in these changes and having discussions related to (possibly dual-intent or not) `config_param::WORKSPACE*` parameters (and values thereof) in the dedicated [PR#509](https://github.com/oneapi-src/oneAPI-spec/pull/509)?

Reply:
No objections from MKL team on removing.

Reply:
We're fine with removing this. We probably won't be adding this back since it's not required.